### ISSUE TITLE
[Feature Request]: Transform data grid improvements

### ISSUE THREAD

**Issue by hansva** [apache/hop#2619]
### What would you like to happen?

Migration: https://issues.apache.org/jira/browse/HOP-3726

Many transforms and dialogs use table components requiring text or numeric entry from cell to cell.

The idea is to make it easier for users to start from the beginning of a row, tab to the right after entry, and keep going until they are ready to go to the next row and press 'enter'.

Imagine a table with four columns.

You type in cell 1:1, pressing tab goes to cell 1:2, you type and decide entering in the first two columns is enough and hit [enter]. The entry becomes live for 2:1 after you pressed [enter]

Basically, tabbing to the next column, lets the app know an enter key will send to the next row.

### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui
